GS: Small fix/addition to DumpVertices for debugging.

TJnotJT opened the pull request:
### Description of Changes
Small changes to the output of DumpVertices to include the m_eq flags from the vertex trace and other formatting things.

### Rationale behind Changes
To make debugging GS issues easier.

### Suggested Testing Steps
Run a GS dump and dump the vertex/register info. The info is written to the "XXXXX_vertex.txt" files where XXXXX is the draw number.

### Did you use AI to help find, test, or implement this issue or feature?
Github copilot for autocompletion.

refractionpcsx2 commented on pcsx2/GS/GSDrawingContext.cpp:
```diff
@@ -226,21 +226,21 @@ void GSDrawingContext::Dump(const std::string& filename)
-		"\tWMS:%u\n"
-		"\tWMT:%u\n"
+		"\tWMS:%s\n"
+		"\tWMT:%s\n"
```
it would be good if it showed the name and maybe the number in brackets? I often forget the order of the two region versions.

TJnotJT replied:
Will do. If it's ok I'll reverse the order so like WMT:0(REPEAT) to make it more uniform.

TJnotJT commented on pcsx2/GS/GSState.cpp:
```diff
@@ -527,19 +527,24 @@ void GSState::DumpVertices(const std::string& filename)
+	file << "\tmin t (s,t,q): " << v2.x << DEL << v2.y << DEL << v2.z << std::endl;
 	v2 = m_vt.m_max.t;
-	file << "\tmax t (x,y,z,w): " << v2.x << DEL << v2.y << DEL << v2.z << DEL << v2.w << std::endl;
+	file << "\tmax t (s,t,q): " << v2.x << DEL << v2.y << DEL << v2.z << std::endl;
```
Would it be ok to rename s,t to u,v since they are at the texel scale? Should be ok to leave q though since both z,w contain q when texture mapping is enabled.

refractionpcsx2 replied:
sure, if you wish

TJnotJT commented on pcsx2/GS/GSDrawingContext.cpp:
```diff
@@ -171,29 +171,29 @@ void GSDrawingContext::Dump(const std::string& filename)
-		"\tX:%u\n"
-		"\tY:%u\n\n",
+		"\tOFX:%u\n"
+		"\tOFXY:%u\n\n",
```
Fixed

TJnotJT commented:
Thanks for the reviews refraction.

For OFX, OFY, I converted to float since I think its more readable this way.

I added new file GSDrawingEnvironment.cpp since I needed to use GSUtil functions.

refractionpcsx2 commented on pcsx2/GS/GSDrawingEnvironment.h:
```diff
@@ -3,6 +3,8 @@
 
 #pragma once
 
+#include "GS/GSDrawingContext.h"
```
did you mean to add this to the cpp file?

refractionpcsx2 replied:
Shouldn't this be in the cpp file?

refractionpcsx2 commented:
> Thanks for the reviews refraction.
> 
> For OFX, OFY, I converted to float since I think its more readable this way.
> 
> I added new file GSDrawingEnvironment.cpp since I needed to use GSUtil functions.

That's fair, since the normal coordinates are converted

TJnotJT commented:
PR has been rebased
